vllm/worker/model_runner.py

```diff
@@ -729,7 +729,10 @@ def _compute_multi_modal_input(self, inter_data: InterDataForSeqGroup,
         mm_kwargs, placeholder_maps = MultiModalPlaceholderMap.from_seq_group(
             seq_group_metadata,
             range(positions[0], positions[0] + len(positions)))
-        if not mm_kwargs:
+
+        # M-RoPE requires mrope_positions even for plain text; return early
+        # when mm_kwargs is empty only if inter_data.is_prompt is False.
+        if not mm_kwargs and not inter_data.is_prompt:
```

Comment:
is this effectively equivalent to checking whether `self.runner.model_config.uses_mrope`?

Reply:
No, the two do not appear to be equivalent. During the prefill stage (the first step), `mm_kwargs` is non-empty, but it becomes an empty dictionary during the next generation phase. The issue arises when processing pure text, where `mm_kwargs` is empty,  and  `mrope_positions` cannot be assigned correct values. 

I think the purpose of this condition is to facilitate an early exit during the generation phase.

![image](https://github.com/user-attachments/assets/b5e0ac4e-082c-41f2-b14d-101d93f7e3c0)

Reply:
oh yeah I get that, I was actually referring to `and not inter_data.is_prompt`, whether it was more accurate to replace with:
```suggestion
        if not mm_kwargs and not self.runner.model_config.uses_mrope:
```

Reply:
self.runner.model_config.uses_mrope` is a static value based on the model config, whereas `inter_data.is_prompt` can differ for each iteration.